### ISSUE TITLE
Amplify Exception for: aws.sdk.kotlin.services.cognitoidentityprovider.model.UserLambdaValidationException

### ISSUE THREAD

**Issue by Lumentus** [aws-amplify/amplify-android#2523]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin - Coroutines

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
    implementation "com.amplifyframework:core-kotlin:$amplify_version"
    implementation "com.amplifyframework:aws-api:$amplify_version"
    implementation "com.amplifyframework:aws-auth-cognito:$amplify_version"
    implementation "com.amplifyframework:aws-storage-s3:$amplify_version"

```

</details>

### Environment information

<details>

```
Does not matter


```

</details>

### Describe the feature request

Currently when a user tries to sign in using cognito and it fails because of a Pre-authentification trigger the signIn Method throws an **UnknownException** with cause: **aws.sdk.kotlin.services.cognitoidentityprovider.model.UserLambdaValidationException**. Since the Kotlin SDK is declared as a developer preview currently and this seems to me like a relatively common case that a user of the Amplify SDK might like to treat specifically, there should be a separate Amplify SDK specific exception, so that we do not have to use a Kotlin SDK type.

### Code Snippet

```java
// Put your code below this line.

```

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by gpanshu** [comment]
You are right, this is indeed a miss. Once we implement this and it gets released I will notify you here.

**Comment by gpanshu** [comment]
This was fixed and released with [Amplify Version 2.12.0](https://github.com/aws-amplify/amplify-android/releases/tag/release_v2.12.0). Thank you for your report.
